### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/mesonbuild-meson/actions/runs/4668669537#summary-12663307665

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/mesonbuild-meson/actions/runs/4668670050#summary-12663310698

### CODE REVIEW

File: .github/workflows/nonnative.yml
Comment (jsoref) [inline .github/workflows/nonnative.yml:1]: This file is being renamed the concept is `non-native`

File: cross/iphone.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This is a cross compilation file from OSX Yosemite to iPhone
 # Apple keeps changing the location and names of files so
-# these might not work for you. Use the googels and xcrun.
+# these might not work for you. Use the googles and xcrun.
```
Comment (jsoref) [inline cross/iphone.txt:3]: Some of meson's code is cute, but I don't see any particular reason to misspell google here to be cute.

File: data/syntax-highlighting/vim/syntax/meson.vim [hunk 2]
```diff
@@ -150,7 +150,7 @@ hi def link mesonEscape		Special
 hi def link mesonNumber		Number
 hi def link mesonBuiltin	Function
 hi def link mesonBoolean	Boolean
-if exists("meson_space_error_higlight")
```
Comment (jsoref) [inline data/syntax-highlighting/vim/syntax/meson.vim:153]: I believe this will change how vim users experience things, as this is the only instance of this, whereas the correct spelling has a couple of instances.

This _doesn't_ _appear_ to be a backwards compatibility shim, although it could be. If it is, I'd be happy to change it to include a comment calling this out...

File: docs/markdown/Release-notes-for-0.40.0.md [hunk 3]
```diff
@@ -119,7 +119,7 @@ qt5_dep = dependency('qt5', modules : 'core', method : 'qmake')
 ## Link whole contents of static libraries
 
 The default behavior of static libraries is to discard all symbols
-that are not not directly referenced. This may lead to exported
+that are not directly referenced. This may lead to exported
```
Comment (jsoref) [inline docs/markdown/Release-notes-for-0.40.0.md:122]: Sometimes double negatives are intentional. usually they're just typos. I'm pretty sure I'm right but not absolutely confident.

File: docs/markdown/Release-notes-for-0.46.0.md [hunk 4]
```diff
@@ -269,7 +269,7 @@ helper = static_library(
 final = shared_library(
   'final',
   ['final.c'],
-  dependencyes : dep,
+  dependencies : dep,
```
Comment (jsoref) [inline docs/markdown/Release-notes-for-0.46.0.md:272]: I don't think this is a problem...

File: docs/markdown/Release-notes-for-0.55.0.md [hunk 5]
```diff
@@ -93,7 +93,7 @@ Meson now supports passing configuration options to CMake and
 overriding certain build details extracted from the CMake subproject.
 
 The new CMake configuration options object is very similar to the
-[[@cfg_data]] object object returned
+[[@cfg_data]] object returned
```
Comment (jsoref) [inline docs/markdown/Release-notes-for-0.55.0.md:96]: I can't tell if this doubling is intentional. I don't think it is.

File: docs/markdown/Release-notes-for-0.62.0.md [hunk 6]
```diff
@@ -19,9 +19,9 @@ directory, that file is loaded by gdb automatically.
 
 ## Print modified environment variables with `meson devenv --dump`
 
-With `--dump` option, all envorinment variables that have been modified are
+With `--dump` option, all environment variables that have been modified are
 printed instead of starting an interactive shell. It can be used by shell
-scripts that wish to setup their environment themself.
+scripts that wish to setup their environment themselves.
```
Comment (jsoref) [inline docs/markdown/Release-notes-for-0.62.0.md:24]: themself has historically been nonstandard and the generally accepted form has been themselves.

There is some shift in modern times to accept `themself` as a word, but...

Anyway, happy to drop this change if people have an opinion.

File: docs/markdown/Vala.md [hunk 7]
```diff
@@ -112,7 +112,7 @@ executable('app_name', sources, dependencies: dependencies)
 ```
 
 Using `[GtkTemplate]` also requires the GTK+ user interface definition
-files to be built in to the binary as GResources. For completeness,
+files to be built into the binary as GResources. For completeness,
```
Comment (jsoref) [inline docs/markdown/Vala.md:115]: `in to`/`into` have been fairly controversial. happy to drop...

File: docs/refman/templates/tags.mustache
Comment (jsoref) [inline docs/refman/templates/tags.mustache:1]: I don't think anything cares and I'm pretty sure it's `tags`...
```diff
-taggs
+tags
```

File: mesonbuild/arglist.py [hunk 8]
```diff
@@ -198,13 +198,13 @@ def _can_dedup(cls, arg: str) -> Dedup:
         """Returns whether the argument can be safely de-duped.
 
         In addition to these, we handle library arguments specially.
-        With GNU ld, we surround library arguments with -Wl,--start/end-gr -> Dedupoup
+        With GNU ld, we surround library arguments with -Wl,--start/end-gr -> Dedup
```
Comment (jsoref) [inline mesonbuild/arglist.py:201]: I'm not particularly confident about this

File: mesonbuild/cmake/traceparser.py [hunk 9]
```diff
@@ -288,7 +288,7 @@ def _gen_exception(self, function: str, error: str, tline: CMakeTraceLine) -> No
         raise CMakeException(f'CMake: {function}() {error}\n{tline}')
 
     def _cmake_set(self, tline: CMakeTraceLine) -> None:
-        """Handler for the CMake set() function in all variaties.
+        """Handler for the CMake set() function in all varieties.
```
Comment (jsoref) [inline mesonbuild/cmake/traceparser.py:291]: probably? (possibly `variables`)

File: mesonbuild/linkers/linkers.py [hunk 10]
```diff
@@ -928,7 +928,7 @@ def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
 
 class CcrxDynamicLinker(DynamicLinker):
 
-    """Linker for Renesis CCrx compiler."""
+    """Linker for Renesas CCrx compiler."""
```
Comment (jsoref) [inline mesonbuild/linkers/linkers.py:931]: Brand, I hope...

File: mesonbuild/modules/keyval.py [hunk 11]
```diff
@@ -55,7 +55,7 @@ def _load_file(path_to_config: str) -> T.Dict[str, str]:
         return result
 
     @noKwargs
-    @typed_pos_args('keyval.laod', (str, mesonlib.File))
+    @typed_pos_args('keyval.load', (str, mesonlib.File))
```
Comment (jsoref) [inline mesonbuild/modules/keyval.py:58]: Scary

File: packaging/createpkg.py [hunk 12]
```diff
@@ -85,7 +85,7 @@ def generate_distribution(self):
         ET.SubElement(root, 'pkg-ref', {'id': self.identifier})
         ET.SubElement(root, 'options', {'customize': 'never',
                                         'require-scripts': 'false',
-                                        'hostArhcitectures': 'x86_64,arm64'})
+                                        'hostArchitectures': 'x86_64,arm64'})
```
Comment (jsoref) [inline packaging/createpkg.py:88]: scary?

File: test cases/common/182 find override/meson.build [hunk 13]
```diff
@@ -18,7 +18,7 @@ assert(tool.found())
 assert(tool.full_path() != '')
 assert(tool.full_path() == tool.path())
 
-# six_meson_exe is an overritten project executable
+# six_meson_exe is an overridden project executable
```
Comment (jsoref) [inline test cases/common/182 find override/meson.build:21]: Sometimes the correct term is `overwritten`, but I _think_ it's an override and as opposed to a file on disk being written over.

File: test cases/common/227 very long command line/codegen.py
Comment (jsoref) [inline test cases/common/227 very long command line/codegen.py:1]:
```diff
-commmand
+command
```

File: test cases/common/235 invalid standard overridden to valid/main.c
Comment (jsoref) [inline test cases/common/235 invalid standard overridden to valid/main.c:1]:
```diff
-overriden
+overridden
```

File: test cases/common/98 subproject subdir/subprojects/sub_implicit/meson.build [hunk 14]
```diff
@@ -10,4 +10,4 @@ sub_implicit_provide2_dep = dep
 # This one is not overridden but the wrap file tells the variable name to use.
 glib_dep = dep
 
-assert(get_option('opt') == 'overriden')
\ No newline at end of file
+assert(get_option('opt') == 'overridden')
```
Comment (jsoref) [inline test cases/common/98 subproject subdir/subprojects/sub_implicit/meson.build]: I try not to mess with missing newlines at eof even though I (and Git/GitHub) am not a fan

File: test cases/failing/129 generator host binary/meson.build [hunk 15]
```diff
@@ -8,7 +8,7 @@ add_languages('c', native : false)
 
 exe = executable('exe', 'exe.c', native : false)
 
-gen = generator(exe, output : '@BASENAME@.c', arguments : ['@INPUT@', '@OUTPU@'])
+gen = generator(exe, output : '@BASENAME@.c', arguments : ['@INPUT@', '@OUTPUT@'])
```
Comment (jsoref) [inline test cases/failing/129 generator host binary/meson.build:11]: scary

File: test cases/failing/59 bad option argument/meson_options.txt [hunk 16]
```diff
@@ -1 +1 @@
-option('name', type : 'string', vaule : 'foo')
+option('name', type : 'string', value_ : 'foo')
```
Comment (jsoref) [inline test cases/failing/59 bad option argument/meson_options.txt:1]: The goal of these tests is to show that a divergence from the expected token will be flagged.

My suggestion is that it doesn't require using a word not in the dictionary. (One could also use `values` instead if one wants a nearby word in the dictionary without introducing punctuation.)

File: test cases/failing/83 dub library/meson.build
Comment (jsoref) [inline test cases/failing/83 dub library/meson.build:1]:
```diff
-test cases/failing/83 dub libray/meson.build
+test cases/failing/83 dub library/meson.build
```

File: test cases/frameworks/1 boost/meson.build [hunk 17]
```diff
@@ -17,7 +17,7 @@ linkdep     = dependency('boost', static: s, modules : ['thread', 'system', 'dat
 testdep     = dependency('boost', static: s, modules : ['unit_test_framework'])
 nomoddep    = dependency('boost', static: s)
 extralibdep = dependency('boost', static: s, modules : ['thread', 'system', 'date_time', 'log_setup', 'log', 'filesystem', 'regex'])
-notfound    = dependency('boost', static: s, modules : ['this_should_not_exist_on_any_systen'], required: false)
+notfound    = dependency('boost', static: s, modules : ['this_should_not_exist_on_any_system'], required: false)
```
Comment (jsoref) [inline test cases/frameworks/1 boost/meson.build:20]:
```diff
-systen
+system
```

**Comment by eli-schwartz** [comment]
From a hasty look at this I think many of the comments you made for changes you weren't clear on, were okay. But I'll need to review this more closely.

n.b. -- one commit per word correction may make sense for ease of refactoring during review, but it will need to be squashed for final merge. We can handle that later.

File: data/shell-completions/zsh/_meson [hunk 18]
```diff
@@ -74,7 +74,7 @@ local -a meson_commands=(
 'configure:configure a project'
 'dist:generate release archive'
 'init:create a new project'
-'install:install one more more targets'
+'install:install one or more targets'
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply data/shell-completions/zsh/_meson:77]: This is the description that zsh users see, so it constitutes a documentation fix. User-facing, yes. Change in effects, no.

File: test cases/common/145 recursive linking/edge-cases/libsto.c [hunk 19]
```diff
@@ -1,8 +1,8 @@
 #include "../lib.h"
 
-int get_builto_value (void);
+int get_built_value (void);
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply test cases/common/145 recursive linking/edge-cases/libsto.c:3]: This is just a lot of inscrutable non-words used as abbreviations, in this case "lib static object" exposes a function called "get static object dep value" which assumes said external static object contains a function called "get built object value".

Of course it's just magic tokens as far as the compiler is concerned, but removing the "o" here doesn't make the function name more correct. Stopping the use of inscrutable abbreviations would.

I think it's more effort than is worth it to care, here. Just drop the hunks.

File: test cases/common/26 find program/meson.build [hunk 20]
```diff
@@ -35,5 +35,5 @@ assert(prog.found(), 'Program version should match')
 prog = find_program('test_subdir.py', required : false)
 assert(not prog.found(), 'Program should not be found')
 
-prog = find_program('test_subdir.py', dirs : ['/donotexist', meson.current_source_dir() / 'scripts'])
+prog = find_program('test_subdir.py', dirs : ['/nonexistent', meson.current_source_dir() / 'scripts'])
```
Comment (eli-schwartz) [inline test cases/common/26 find program/meson.build:38]: Technically not wrong before, but okay.

File: test cases/frameworks/14 doxygen/doc/Doxyfile.in [hunk 21]
```diff
@@ -278,7 +278,7 @@ OPTIMIZE_OUTPUT_VHDL   = NO
 # parses. With this tag you can assign which parser to use for a given
 # extension. Doxygen has a built-in mapping, but you can override or extend it
 # using this tag. The format is ext=language, where ext is a file extension, and
-# language is one of the parsers supported by doxygen: IDL, Java, Javascript,
+# language is one of the parsers supported by doxygen: IDL, Java, JavaScript,
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply test cases/frameworks/14 doxygen/doc/Doxyfile.in:281]: I would mark this file as thirdparty and avoid checking it, yeah.

File: unittests/allplatformstests.py [hunk 22]
```diff
@@ -1760,10 +1760,10 @@ def test_pkgconfig_static(self):
         since system libraries -lm will never be found statically.
         https://github.com/mesonbuild/meson/issues/2785
         '''
-        (cc, stlinker, objext, shext) = self.detect_prebuild_env()
+        (cc, stlinker, object, shext) = self.detect_prebuild_env()
```
Comment (eli-schwartz) [inline unittests/allplatformstests.py:1763]: This is an ext for an obj, so it should be reverted.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
Mostly good, I do have a few comments.

On renaming things from e.g. "Github to GitHub" and "Javascript to JavaScript" I'm -0, I don't really consider either one to be wrong.

File: mesonbuild/interpreter/compiler.py [hunk 23]
```diff
@@ -405,7 +405,7 @@ def has_type_method(self, args: T.Tuple[str], kwargs: 'CommonKW') -> bool:
         KwargInfo('guess', (int, NoneType)),
         *_COMMON_KWS,
     )
-    def compute_int_method(self, args: T.Tuple[str], kwargs: 'CompupteIntKW') -> int:
+    def compute_int_method(self, args: T.Tuple[str], kwargs: 'ComputeIntKW') -> int:
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply mesonbuild/interpreter/compiler.py:408]: Meson has no public API, so as long as all internal uses are renamed to match then this can be safely renamed.

File: packaging/createpkg.py (hunk 12 above)
Comment (eli-schwartz) [reply packaging/createpkg.py:88]: Probably this would be a functional change for the macOS packages, although I don't know how many people actually use those.

File: test cases/common/98 subproject subdir/subprojects/sub_implicit/meson.build (hunk 14 above)
Comment (eli-schwartz) [reply test cases/common/98 subproject subdir/subprojects/sub_implicit/meson.build]: Purge missing newlines with fire. :P

File: packaging/createpkg.py (hunk 12 above)
Comment (jsoref) [reply packaging/createpkg.py:88]: I installed `meson` via `brew install meson` when i was trying to figure out the `zsh` thing... I'm not sure if that counts.

File: test cases/frameworks/7 gnome/gir/meson.build [hunk 24]
```diff
@@ -4,7 +4,7 @@ libsources = ['meson-sample.c', 'meson-sample.h']
 lib2sources = ['meson-sample2.c', 'meson-sample2.h']
 
 gen_source = custom_target(
-  'meson_smaple3.h',
+  'meson_sample3.h',
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply test cases/frameworks/7 gnome/gir/meson.build:7]: It's used as the descriptive name in the `ninja` status lines, but otherwise ignored.

File: packaging/createpkg.py (hunk 12 above)
Comment (eli-schwartz) [reply packaging/createpkg.py:88]: It doesn't, brew uses pip basically. createpkg.py creates something that could be added to the macOS App Store or whatever.

File: mesonbuild/cmake/traceparser.py (hunk 9 above)
Comment (eli-schwartz) [reply mesonbuild/cmake/traceparser.py:291]: I think you're correct, here.

File: mesonbuild/modules/keyval.py (hunk 11 above)
Comment (eli-schwartz) [reply mesonbuild/modules/keyval.py:58]: Well, it's just the text for the error message if you use invalid arguments to the load() function. Pretty silly but non-impactful other than as a source of confusion.

File: test cases/failing/75 non-ascii in ascii encoded configure file/meson.build
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply test cases/failing/75 non-ascii in ascii encoded configure file/meson.build:1]: Makes sense.

File: test cases/frameworks/4 qt/meson.build [hunk 25]
```diff
@@ -65,7 +65,7 @@ foreach qt : ['qt4', 'qt5', 'qt6']
     extra_cpp_args = []
     if meson.is_unity()
       extra_cpp_args += '-DUNITY_BUILD'
-      prep_rcc = qtmodule.preprocess(qt + '_unity_ressource', qresources : ['stuff.qrc', 'stuff2.qrc'], method : get_option('method'))
+      prep_rcc = qtmodule.preprocess(qt + '_unity_resource', qresources : ['stuff.qrc', 'stuff2.qrc'], method : get_option('method'))
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply test cases/frameworks/4 qt/meson.build:68]: This is used as the generated output filename for Qt resource files (and as internal ids).

It makes no difference what name you use, so fixing the typo is fine and doesn't even require syncing a token in another file.

File: docs/markdown/Release-notes-for-0.60.0.md [hunk 26]
```diff
@@ -357,7 +357,7 @@ are found, and silently continue if Visual Studio activation fails.
 
 `meson setup --vsenv` command line argument can now be used to force Visual Studio
 activation even when other compilers are found. It also make Meson abort with an
-error message when activation fails. This is especially useful for Github Action
+error message when activation fails. This is especially useful for GitHub Action
```
[the remark this answers is not in the document]
Comment (jsoref) [reply docs/markdown/Release-notes-for-0.60.0.md:360]: Happy to fix.

In general, my tool looks for misspelled words, that I happen to be able to add patterns to catch additional things is a happy coincidence.

A quick skim shows that _sometimes_ people do mean a singular GitHub Action, and thus I probably couldn't easily add a generic rule:
https://grep.app/search?q=Github%20Action&case=true&words=true
(Each project I scan enables me to either improve my software or the software's default configuration -- this repository found some edges in how I handle repositories w/ `.gitignore` which was interesting, so I'm quite thankful that I took the opportunity to revisit it.)

**Comment by eli-schwartz** [comment]
For functional changes that fix a runtime bug (the visual studio openmp thing, vim support files, etc.) I'd prefer to see distinct commits that fix the bug, describing the bug that is being fixed (and crediting the spell checker, I guess).

The rest should be folded into a commit for fixing various spelling errors.

File: docs/markdown/Release-notes-for-0.60.0.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply docs/markdown/Release-notes-for-0.60.0.md:360]: Neat. :)

File: .github/workflows/unusedargs_missingreturn.yml [hunk 27]
```diff
@@ -23,7 +23,7 @@ on:
     - "test cases/linuxlike/**"
     - "test cases/objc/**"
     - "test cases/objcpp/**"
-    - "test caes/windows/**"
```
[the remark this answers is not in the document]
Comment (jsoref) [reply .github/workflows/unusedargs_missingreturn.yml:26]: Would you want this to be backported? It isn't technically a functional change, but it's a coverage change that might protect against other changes...

[the remark this answers is not in the document]
Comment (eli-schwartz) [reply .github/workflows/unusedargs_missingreturn.yml:26]: It's not functional per se, but it's fixing something other than grammar. So I'd probably keep it as a distinct commit for purposes of navigating the `git log`, even though it's most likely not going to be backported.

[the remark this answers is not in the document]
Comment (eli-schwartz) [reply .github/workflows/unusedargs_missingreturn.yml:26]: (In general, the common/ directory changes very often so we know this is passing at git master, hence coverage is fine.)

**Comment by eli-schwartz** [comment]
@nirbheek some of these commits should be manually backported, particularly see:
- commit 2: packaging hostArchitectures
- commit 3: the vim syntax highlighting one
- commit 7: OpenMPSupport
